src/lib/format/l20n/ast/parser.js

```diff
@@ -75,7 +75,10 @@ class ParseContext {
     }
 
     const ch = this._source.charAt(this._index);
-    const value = this.getValue(ch, index === undefined);
+    /*jshint -W018 */
+    const hasIndex = !(index === undefined);
+    const isRequired = hasIndex;
```

Comment:
Is this about the value being required?  If so, can you call this `valueRequired`?